golang/commoncrawl/commoncrawl.go

```diff
@@ -0,0 +1,67 @@
+package commoncrawl
+
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+)
+
+const (
+	BaseURL   = "https://data.commoncrawl.org"
+	CrawlPath = "cc-index/collections/CC-MAIN-2024-30/indexes"
+)
+
+type Downloader interface {
+	DownloadAndUnzip(url string, start, length int) ([]byte, error)
+}
+
+type CCDownloader struct {
+	baseURL string
+}
+
+func NewCCDownloader(baseURL string) *CCDownloader {
```

Comment:
Shouldn't we return the interface?

```suggestion
func NewCCDownloader(baseURL string) Downloader {
```

Or do we want to do this on purpose to see how the candidate handles it?